File: bot.go
package main

import (
	"bytes"
	"encoding/json"
	"flag"
	"fmt"
	"io"
	"log"
	"math/rand"
	"net/http"
	"os"
	"strconv"
	"time"
)

type Message struct {
	Nickname string `json:"nickname"`
	Text     string `json:"text"`
}

func generateRandomString(n int) string {
	const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
	b := make([]byte, n)
	for i := range b {
		b[i] = letters[rand.Intn(len(letters))]
	}
	return string(b)
}

func generateNickname() string {
	firstNames := []string{"Cool", "Super", "Mega", "Ultra", "Epic", "Lucky", "Silent", "Mighty"}
	lastNames := []string{"Hero", "Warrior", "Ninja", "Samurai", "Wizard", "Knight", "Ranger", "Hunter"}

	return fmt.Sprintf("%s%s%d", firstNames[rand.Intn(len(firstNames))], lastNames[rand.Intn(len(lastNames))], rand.Intn(1000))
}

func sendMessage(url string) {
	message := Message{
		Nickname: generateNickname(),
		Text:     generateRandomString(32),
	}

	jsonData, err := json.Marshal(message)
	if err != nil {
		log.Fatalf("JSON serialization error: %v", err)
	}

	resp, err := http.Post(url, "application/json", bytes.NewBuffer(jsonData))
	if err != nil {
		log.Printf("Sending error: %v", err)
		return
	}
	defer func(Body io.ReadCloser) {
		err := Body.Close()
		if err != nil {
			return
		}
	}(resp.Body)

	fmt.Printf("Server response: %s\n", resp.Status)
}

func main() {
	messagePerMinute := os.Getenv("BOT_MESSAGES_PER_MINUTE")
	rate, err := strconv.Atoi(messagePerMinute)
	if err != nil {
		rate = 60
	}
	url := os.Getenv("BOT_SERVICE_URL")
	if url == "" {
		url = "http://app:3000/api/messages/"
	}
	flag.Parse()

	interval := time.Minute / time.Duration(rate)

	for {
		sendMessage(url)
		time.Sleep(interval)
	}
}

File: bot/bot.py
import requests
import random
import time
import argparse
import os

def generate_nickname():
    first_names = ["Cool", "Super", "Mega", "Ultra", "Epic", "Lucky", "Silent", "Mighty"]
    last_names = ["Hero", "Warrior", "Ninja", "Samurai", "Wizard", "Knight", "Ranger", "Hunter"]
    nickname = f"{random.choice(first_names)}{random.choice(last_names)}{random.randint(0, 999)}"
    return nickname

def send_message(url):
    message = {
        "nickname": generate_nickname(),
        "text": "random"
    }
    response = requests.post(url, json=message)
    print(f"Message sent. Status: {response.status_code}")

def main():
    enable = os.environ.get("BOT_ENABLE", "false")

    if enable.lower() != "true":
        print("Bot is disabled")
        return

    rate = os.environ.get("BOT_MESSAGES_PER_MINUTE", 60)
    url = os.environ.get("BOT_SERVICE_URL", "http://localhost:3000/api/messages")

    interval = 60.0 / int(rate)

    while True:
        try:
            send_message(url)
        except Exception as e:
            print(f"Error: {e}")
        time.sleep(interval)

if __name__ == "__main__":
    main()
